tests/components/zha/test_config_flow.py

```diff
@@ -1,125 +1,237 @@
+@patch(
+    "homeassistant.components.zha.config_flow.detect_radios",
+    return_value={CONF_RADIO_TYPE: "test_radio"},
+)
+async def test_user_flow(detect_mock, hass):
+    """Test user flow -- radio detected."""
 
-    assert result["type"] == "create_entry"
-    assert result["title"] == "/dev/ttyUSB1"
-    assert result["data"] == {"usb_path": "/dev/ttyUSB1", "radio_type": "ezsp"}
+    port = com_port()
+    port_select = f"{port}, s/n: {port.serial_number} - {port.manufacturer}"
 
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={CONF_SOURCE: SOURCE_USER},
+        data={zigpy.config.CONF_DEVICE_PATH: port_select},
+    )
+    assert result["type"] == RESULT_TYPE_CREATE_ENTRY
+    assert result["title"].startswith(port.description)
+    assert result["data"] == {CONF_RADIO_TYPE: "test_radio"}
+    assert detect_mock.await_count == 1
+    assert detect_mock.await_args[0][0] == port.device
+
+
+@patch("serial.tools.list_ports.comports", MagicMock(return_value=[com_port()]))
+@patch(
+    "homeassistant.components.zha.config_flow.detect_radios", return_value=None,
+)
+async def test_user_flow_not_detected(detect_mock, hass):
+    """Test user flow, radio not detected."""
+
+    port = com_port()
+    port_select = f"{port}, s/n: {port.serial_number} - {port.manufacturer}"
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={CONF_SOURCE: SOURCE_USER},
+        data={zigpy.config.CONF_DEVICE_PATH: port_select},
+    )
 
-async def test_user_flow_existing_config_entry(hass):
-    """Test if config entry already exists."""
-    MockConfigEntry(domain=DOMAIN, data={"usb_path": "/dev/ttyUSB1"}).add_to_hass(hass)
-    flow = config_flow.ZhaFlowHandler()
-    flow.hass = hass
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["step_id"] == "pick_radio"
+    assert detect_mock.await_count == 1
+    assert detect_mock.await_args[0][0] == port.device
 
-    result = await flow.async_step_user()
 
-    assert result["type"] == "abort"
+async def test_user_flow_show_form(hass):
+    """Test user step form."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={CONF_SOURCE: SOURCE_USER},
+    )
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
 
 
-async def test_import_flow(hass):
-    """Test import from configuration.yaml ."""
-    flow = config_flow.ZhaFlowHandler()
-    flow.hass = hass
+async def test_user_flow_manual(hass):
+    """Test user flow manual entry."""
 
-    result = await flow.async_step_import(
-        {"usb_path": "/dev/ttyUSB1", "radio_type": "xbee"}
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={CONF_SOURCE: SOURCE_USER},
+        data={zigpy.config.CONF_DEVICE_PATH: config_flow.CONF_MANUAL_PATH},
     )
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["step_id"] == "pick_radio"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5462445)